src/features/SpellCheckerProvider.ts

```diff
@@ -202,8 +202,10 @@ export default class SpellCheckerProvider implements vscode.CodeActionProvider
 
 		text = this.processUserIgnoreRegex( text );
 
-		// remove pandoc yaml header
-		text = text.replace( /---(.|\n)*\.\.\./g, ' ' );
```

Comment:
You'll find more info on this issue there: https://github.com/adrienjoly/vscode-spellchecker/issues/2#issuecomment-288131985

You can cancel this pull request if you want, tho :-)